File: jimmer-example/spring-boot-with-jimmer-multiple-datasource/src/main/java/org/jimmer/jimmerdatasouce/domain/BaseEntity.java
package org.jimmer.jimmerdatasouce.domain;

import org.babyfish.jimmer.sql.MappedSuperclass;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;

/**
 * 公共实体 @MappedSuperclass 修饰，全部字段为可空
 * @author jhlz
 * @version x.x.x
 */
@MappedSuperclass
public interface BaseEntity {
    @Nullable
    LocalDateTime createTime();
    @Nullable
    LocalDateTime updateTime();
    @Nullable
    String createBy();
    @Nullable
    String updateBy();
    @Nullable
    String remark();
}

File: jimmer-example/quarkus-with-jimmer-kt/src/main/kotlin/org/jimmer/domain/BaseEntity.kt
package org.jimmer.domain

import org.babyfish.jimmer.sql.MappedSuperclass
import java.time.LocalDateTime

/**
 * @author jhlz
 * @version x.x.x
 */
@MappedSuperclass
interface BaseEntity {

    val createTime: LocalDateTime

    val updateTime: LocalDateTime

    val createBy: String?

    val updateBy: String?
}